### PR TITLE
FY 2023 GitHub Data Report

### LINKED ISSUE #4: Develop report for SET GitHub contributions during fiscal year 2023
Date range: 7/1/2022 - 6/30/2023

### PR DESCRIPTION
**Pull request by d33bs**
Adds fiscal year 2023 data reporting from GitHub with reusable notebook.

Closes #4

### CODE REVIEW

File: set_effort_analysis/github_contributions_fy_2023.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+# In[5]:
+
+
+# set parameters for gathering data
+params = {
+    # get issues with all states
+    "state": "all",
+    # get only the first page of 400 results
+    "page": "1",
+    "per_page": 400,
```
Comment (vincerubinetti) [inline set_effort_analysis/github_contributions_fy_2023.py]: I think the max value here is 100?
https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28

So for certain orgs that have a ton of repos, like the greenelab, you might be missing some. You might have to loop through the pagination here.

**Review by vincerubinetti — APPROVED** [review]
This looks good as far as what data it is providing. I think these numbers will largely be glossed over in presentations and such, so we'd probably want to wait until they request more specific things before taking the time to implement more data gathering.

**Comment by falquaddoomi** [comment]
I agree with Vince; these numbers are good enough for a short presentation. Thanks for gathering them!

Regarding the pagination comment, @d33bs, you might consider using https://github.com/PyGithub/PyGithub rather than rolling your own implementation, as it handles iterating through paginated requests for you. If there's another reason you'd rather write it yourself, feel free, of course.

**Review by falquaddoomi — APPROVED** [review]

File: set_effort_analysis/github_contributions_fy_2023.py (hunk 1 above)
Comment (vincerubinetti) [reply set_effort_analysis/github_contributions_fy_2023.py]: Looks like greenelab, monarch-init, and krishnanlab are the only ones that go over 100. And the first two go over significantly.

**Comment by d33bs** [comment]
Thanks @vincerubinetti and @falquaddoomi for the reviews! I made updates to use PyGithub along with a few other adjustments based on your comments. Merging for now.

Note: In the future to avoid rate limits with Github it may make sense to change this approach to be user-centric instead of org-centric; that is, parsing issue and pull request data from team members as users and filtering down to the relevant organizations. Doing this would likely reduce the amount of data being parsed, increasing the speed of the report and reducing the risk of rate limit exceptions.

File: set_effort_analysis/github_contributions_fy_2023.py (hunk 1 above)
Comment (d33bs) [reply set_effort_analysis/github_contributions_fy_2023.py]: Thanks for this comment, great point! This has been addressed using PyGithub's built-in pagination.

**Pull request merged**